lib/rust/flexer/src/generate.rs

```
@@ -0,0 +1,557 @@
+) -> Result<Block,GenError> {
+    if target_state == Identifier::INVALID {
+        match maybe_state {
+            None => {
+                Ok(parse_quote! {{
+                    StageStatus::ExitFail
+                }})
+            },
+            Some(rule_exec) => {
+                let rule:Expr = match parse_str(rule_exec.code.as_str()) {
+                    Ok(rule) => rule,
+                    Err(_)   => return Err(GenError::BadExpression(rule_exec.code.clone()))
+                };
+                if rules_overlap {
+                    Ok(parse_quote! {{
+                        let rule_bookmark    = self.bookmarks.rule_bookmark;
+                        let matched_bookmark = self.bookmarks.matched_bookmark;
+                        self.bookmarks.rewind(rule_bookmark,reader);
+                        self.current_match = reader.pop_result();
+                        self.#rule(reader);
+                        self.bookmarks.bookmark(matched_bookmark,reader);
+                        StageStatus::ExitSuccess
+                    }})
+                } else {
+                    Ok(parse_quote! {{
+                        let matched_bookmark = self.bookmarks.matched_bookmark;
+                        self.current_match   = reader.pop_result();
+                        self.#rule(reader);
+                        self.bookmarks.bookmark(matched_bookmark,reader);
+                        StageStatus::ExitSuccess
+                    }})
+                }
+            }
+        }
+    } else {
+        let target_state_has_no_rule = match maybe_state {
+            Some(state) => if !dfa.has_rule_for(target_state) {
+                dfa.callbacks[target_state.id] = Some(state.clone());
+                has_overlaps.insert(target_state.id,true);
+                true
+            } else {
+                false
+            },
+            None => false
+        };
+
+        let state_id:Lit = str_to_lit(format!("{}",target_state.id))?;
```

Won't using a `Literal` or evn `usize` value directly in the `parse_quote!` work?
I don't remember the rules exactly but all these occurrences of "pretty-print -> parse -> quote" look like something spurious. I can understand this when dealing with externally provided string that might not be a valid identifier (then we want to fail early rather than generate broen code) but this is not possible with number literals.